pkg/networkserver/grpc_gsns.go

```diff
@@ -951,23 +999,8 @@ func (ns *NetworkServer) handleDataUplink(ctx context.Context, up *ttnpb.UplinkM
 			}
 
 			if !stored.CreatedAt.Equal(matched.Device.CreatedAt) || !stored.UpdatedAt.Equal(matched.Device.UpdatedAt) {
-				rematched, err := ns.matchAndHandleDataUplink(up, true, contextualEndDevice{
-					Context:   ctx,
-					EndDevice: stored,
-				})
-				if err != nil {
-					return nil, nil, errOutdatedData.WithCause(err)
-				}
-				matched = rematched
-
-				ctx = matched.Context
-				pld.FullFCnt = matched.FCnt
-				up.DeviceChannelIndex = uint32(matched.ChannelIndex)
-				up.Settings.DataRateIndex = matched.DataRateIndex
-				logger = log.FromContext(ctx).WithFields(log.Fields(
-					"data_rate_index", up.Settings.DataRateIndex,
-					"device_channel_index", up.DeviceChannelIndex,
-				))
+				// TODO: Rematch device (ISSUE REF)
```

Comment:
I will still address this and make some changes to the matching utility function(s)